ToolChain/Debugger/Debugger.py

```diff
@@ -589,8 +623,26 @@ def addBreakPoint(self, address=None):
                 self.debugLog("Error while processing address or symbol {}".format(address,))
                 return
 
-        self.breakPoints.append(address)
+        if address in self.breakPoints:
```

Comment:
What about changing the code around to make it simpler?
```suggestion
        if address not in self.breakPoints:
            self.breakPoints.append(address)
            self.breakPoints.sort()
            self.writeBreakPointFile()
```

Reply:
Also, nice catch of finding that bug ;)